Update documentation to v2.0.0 (#50)
eProsima/Micro-XRCE-DDS-docs: docs/installation.rst

=========================

> To compile and install the Client and the Agent modules, we use CMake.

Reviewer: ```suggestion
To compile and install the Client and the Agent modules, CMake is used.
```

    $ cmake --build .
    $ cmake --build . --target install

.. note::
    The *eProsima Micro XRCE-DDS Agent* can be configured at compile-time via several CMake definitions.
    Find them listed in the :ref:`agent_configuration` section of the :ref:`micro_xrce_dds_agent_label` page.

Now you have the executable *eProsima Micro XRCE-DDS Agent* installed in your system. Before running it, you need to add
:code:`/usr/local/lib` to the dynamic loader-linker directories. ::

    sudo ldconfig /usr/local/lib/

.. important::
    The *eProsima Micro XRCE-DDS Agent* executable comes with a rich CLI.
    Find out all the options offered by this CLI when running the *Agent* in the :ref:`agent_cli` section of the
    :ref:`micro_xrce_dds_agent_label` page. 

> .. TODO: Create a section on snap installation here.

Reviewer: @jamoralp TODO

Reviewer 2: done

.. important::
    The *eProsima Micro XRCE-DDS Agent* executable comes with a rich CLI.
    Find out all the options offered by this CLI when running the *Agent* in the :ref:`agent_cli` section of the
    :ref:`micro_xrce_dds_agent_label` page.

Installation from Snap package
******************************

    If you wish, you can also install the *eProsima Micro XRCE-DDS Agent* as a `Snap package. <https://snapcraft.io/micro-xrce-dds-agent>`_

    To do so, simply execute ``sudo snap install micro-xrce-dds-agent`` in your Linux console.

>     This will download the Snap package corresponding to the ``stable`` version, that is, *master* branch on GitHub.

Reviewer: *master* branch on GitHub -> the *master* branch on GitHub

>     You can add the ``--edge`` flag to the installation command, just in case you want to download the Snap image corresponding to the *develop* branch on the upstream source code repository.

Reviewer: on the upstream source code repository -> from the upstream source code repository

.. _install_agent_docker:

Using the provided Docker image
*******************************

>     The *eProsima Micro XRCE-DDS Agent* can also be launched directly from its dedicated `Docker image. <https://hub.docker.com/r/eprosima/micro-xrce-dds-agent>`_

Reviewer: Sentence missing a final full stop.

Reviewer 2: The full stop is included in the html reference :)

    You can pull the image to your local machine by executing ``docker pull eprosima/micro-xrce-dds-agent:<tag> <<args>>``, with *tag* being one of the following options:

    * ``stable``: `Micro XRCE-DDS Agent master <https://github.com/eProsima/Micro-XRCE-DDS-Agent/tree/master>`_ branch.
    * ``latest``: `Micro XRCE-DDS Agent develop <https://github.com/eProsima/Micro-XRCE-DDS-Agent/tree/develop>`_ branch)
    * ``vM.m.p``: `Micro XRCE-DDS Agent tagged <https://github.com/eProsima/Micro-XRCE-DDS-Agent/tags>`_ versions, with the *Major*, *minor*, *patch* structure.

>     The accepted arguments for ``<<args>>`` are exactly the same ones which are listed in the :ref:`agent_cli` section.

Reviewer: same ones -> same

There are several CMake definitions for configuring the build of the client library at compile time.
You can found them in :ref:`micro_xrce_dds_client_label` page under `Configuration` section.

For building your Client app, you need to build against the following libs: ::

    gcc <your_main.c> -lmicrocdr -lmicroxrcedds_client

Installing the XRCE-DDS Gen standalone
--------------------------------

Clone the project from GitHub: ::

    $ sudo apt install git openjdk-8-jdk gradle
    $ git clone https://github.com/eProsima/Micro-XRCE-DDS-Gen.git
    $ cd Micro-XRCE-DDS-Gen
    $ git submodules init
    $ git submodules update
    $ gradle build -Dbranch=v1.2.5  

> You will have the XRCE-DDS available as: ::

Reviewer: ```suggestion
You will have the *Micro XRCE-DDS-Gen* tool available as: ::
```

    gcc <your_main.c> -lmicrocdr -lmicroxrcedds_client

.. _install_client_docker:

Using the provided Docker image
*******************************

>     The *eProsima Micro XRCE-DDS Client* comes with a `Docker image <https://hub.docker.com/r/eprosima/micro-xrce-dds-agent>`_ where the library is installed, as well as the provided examples, so users can easily execute them.

Reviewer: where the library is installed together with the examples, so they can easily be executed by the users.

    You can pull the image to your local machine by executing ``docker pull eprosima/micro-xrce-dds-client:<tag> <<args>>``, with *tag* being one of the following options:

    * ``stable``: `Micro XRCE-DDS Client master <https://github.com/eProsima/Micro-XRCE-DDS-Client/tree/master>`_ branch.
    * ``latest``: `Micro XRCE-DDS Client develop <https://github.com/eProsima/Micro-XRCE-DDS-Client/tree/develop>`_ branch)
    * ``vM.m.p``: `Micro XRCE-DDS Client tagged <https://github.com/eProsima/Micro-XRCE-DDS-Client/tags>`_ versions, with the *Major*, *minor*, *patch* structure.

>     The accepted arguments for ``<<args>>`` are the examples' executable names, along with the arguments required for each example to work.

Reviewer: along with -> followed by

.. note::
    If you want to install the *eProsima Micro XRCE-DDS* examples, you can add :code:`-DUXRCE_BUILD_EXAMPLES=ON`
    to the :code:`cmake ..` command-line options. This flag will enable the compilation of the examples.

.. _install_agent_client_docker:

Using the provided Docker image
*******************************

>     *eProsima Micro XRCE-DDS* is also available as a whole package in a `Docker image. <https://hub.docker.com/r/eprosima/micro-xrce-dds-agent>`_

Reviewer: Missing full stop at the end.

Reviewer 2: Same as before, full stop included in the html reference

>     Within this Docker image, the *Micro XRCE-DDS Agent* standalone application and library are installed, as well as the *Micro XRCE-DDS Client* library and built-in examples.

Reviewer: Within -> With

Reviewer 2: well, actually the installation is done inside the Docker. Wouldn't it make more sense to keep it as within?
